apiserver/facades/agent/provisioner/provisioner.go

```diff
@@ -1578,6 +1578,9 @@ func (api *ProvisionerAPI) Remove(ctx context.Context, args params.Entities) (pa
 			continue
 		}
 
+		// TODO (stickupkid): We should workout whether we want to force
+		// removal based on the current state of the removal process. For now
+		// we always pass false.
 		err = api.removalService.DeleteMachine(ctx, machineUUID)
```

Comment:
This shouldn't be done here. This should be some action like advancing the machine life to dead, allowing a removal job to complete.

This is diluting how the removal domain is supposed to work. If it's done right, we don't need these was-job-forced methods to find things out out-of-band.